command/scripting.go

```diff
@@ -0,0 +1,375 @@
+
+	// Add defined expansions to the environment of the process
+	// that's launched.
+	AddExpansionsToEnv bool `mapstructure:"add_expansions_to_env"`
+
+	// IncludeExpansionsInEnv allows users to specify a number of
+	// expansions that will be included in the environment, if
+	// they are defined. It is not an error to specify expansions
+	// that are not defined in include_expansions_in_env.
+	IncludeExpansionsInEnv []string `mapstructure:"include_expansions_in_env"`
+
+	// Silent, if set to true, prevents shell code/output from being
+	// logged to the agent's task logs. This can be used to avoid
+	// exposing sensitive expansion parameters and keys.
+	Silent bool `mapstructure:"silent"`
+
+	// SystemLog if set will write the shell command's output to the system logs, instead of the
+	// task logs. This can be used to collect diagnostic data in the background of a running task.
+	SystemLog bool `mapstructure:"system_log"`
+
+	// WorkingDir is the working directory to start the shell in.
+	WorkingDir string `mapstructure:"working_dir"`
+
+	// IgnoreStandardOutput and IgnoreStandardError allow users to
+	// elect to ignore either standard error and/or standard output.
+	IgnoreStandardOutput bool `mapstructure:"ignore_standard_out"`
+	IgnoreStandardError  bool `mapstructure:"ignore_standard_error"`
+
+	// RedirectStandardErrorToOutput allows you to capture
+	// standard error in the same stream as standard output. This
+	// improves the synchronization of these streams.
+	RedirectStandardErrorToOutput bool `mapstructure:"redirect_standard_error_to_output"`
+
+	// ContinueOnError determines whether or not a failed return code
+	// should cause the task to be marked as failed. Setting this to true
+	// allows following commands to execute even if this shell command fails.
+	ContinueOnError bool `mapstructure:"continue_on_err"`
+
+	base
+}
+
+func subprocessScriptingFactory() Command {
+	return &scriptingExec{}
+}
+
+func (c *scriptingExec) Name() string { return "subprocess.scripting" }
+func (c *scriptingExec) ParseParams(params map[string]interface{}) error {
+	err := mapstructure.Decode(params, c)
+	if err != nil {
+		return errors.Wrapf(err, "error decoding %s params", c.Name())
+	}
+
+	if c.Command != "" {
+		if c.Script != "" || len(c.Args) > 0 {
+			return errors.New("must specify command as either arguments or a command string but not both")
+		}
+
+		c.Args, err = shlex.Split(c.Command)
+		if err != nil {
+			return errors.Wrapf(err, "problem parsing %s command", c.Name())
+		}
+	}
+
+	if c.Script == "" && len(c.Args) == 0 {
+		return errors.New("must specify either a script or a command")
+	}
+	if c.Script != "" && len(c.Args) > 0 {
+		return errors.New("must specify either a script or a command, but not both")
+	}
+
+	if c.CacheDurationSeconds < 1 {
+		c.CacheDurationSeconds = 10
+	}
+
+	if c.Silent {
+		c.IgnoreStandardError = true
+		c.IgnoreStandardOutput = true
+	}
+
+	if c.IgnoreStandardOutput && c.RedirectStandardErrorToOutput {
```

Comment:
I'd suggest separate case for `c.Silent == true && c.RedirectStandardErrorToOutput` for clarity.

Reply:
this is 100% the same as subprocess exec.